### PR TITLE
Fix NoneType errors in EditorTest

### PR DESCRIPTION
**Pull request by Kadino**
Reduces EditorTest NoneType errors, remove dead code, update docstrings, fix pep8 warnings.

Addresses a failure seen in https://github.com/o3de/o3de/issues/7657 which unexpectedly removed test reporting from an AR failure.

Signed-off-by: sweeneys <sweeneys@amazon.com>

### CODE REVIEW

**Comment by Kadino** [comment]
Fixed some initialized-as-mutable attributes, but avoided addressing all data lifetime concerns. There are a number of class-stored attributes that may cause problems to existing scripts if they are move to instance-level ones. Most of these concerns will require a larger change, such as changes to `cmdline_args` also needing to modify how tests get collated into separate batches. Fixing this will later be necessary if tests get batched across a global suite, instead of only per CTest module.

File: Tools/LyTestTools/ly_test_tools/o3de/editor_test.py [hunk 1]
```diff
@@ -1072,14 +1122,16 @@ def run(request, workspace, extra_cmdline_args):
 
         save_asset_logs = False
         for result in results_per_thread:
+            if result is None:
+                result = Result.Unknown(test_spec=EditorSharedTest)
```
Comment (evanchia-ly-sdets) [inline Tools/LyTestTools/ly_test_tools/o3de/editor_test.py:1126]: Here and for the rest of the same- Should we add some extra information for this Result? Could pass in something useful in the extra_info arg to be printed.

**Review by evanchia-ly-sdets — CHANGES_REQUESTED** [review]

File: Tools/LyTestTools/ly_test_tools/o3de/editor_test.py [hunk 2]
```diff
@@ -1016,16 +1057,21 @@ def run(request, workspace, extra_cmdline_args):
...
             # If at least one test did not pass, save assets with errors and warnings
             if save_asset_logs:
                 editor_utils.save_failed_asset_joblogs(workspace)
 
-    def _run_parallel_batched_tests(self, request: Request, workspace: AbstractWorkspace, editor: Editor, editor_test_data: TestData,
-                                    test_spec_list: list[EditorSharedTest], extra_cmdline_args: list[str] = []) -> None:
+    def _run_parallel_batched_tests(self, request: _pytest.fixtures.FixtureRequest,
+                                    workspace: ly_test_tools._internal.managers.workspace.AbstractWorkspaceManager,
+                                    editor: ly_test_tools.launchers.platforms.base.Launcher, editor_test_data: TestData,
+                                    test_spec_list: list[EditorSharedTest], extra_cmdline_args: list[str] = None) -> None:
```
[the remark this answers is not in the document]
Comment (Kadino) [reply Tools/LyTestTools/ly_test_tools/o3de/editor_test.py:1074]: It's at least not a pep8 violation, but I'm happy to make this change

[the remark this answers is not in the document]
Comment (Kadino) [reply Tools/LyTestTools/ly_test_tools/o3de/editor_test.py:1074]: Appears that removing the spaces for type-hinted parameter values creates a pep8 warning, so I'll not make this change

**Review by evanchia-ly-sdets — APPROVED** [review]

File: Tools/LyTestTools/ly_test_tools/o3de/editor_test.py [hunk 3]
```diff
@@ -599,7 +601,9 @@ def will_run(item):
             try:
                 skipping_pytest_runtest_setup(item)
                 return True
-            except Exception:  # intentionally broad
+            except (Warning, Exception, _pytest.outcomes.OutcomeException) as ex:
+                # intentionally broad to avoid events other than system interrupts
+                warnings.warn(f"Test deselected from execution queue due to {ex}")
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/o3de/editor_test.py:606]: Apparently the previously broader exception clause was required, particularly for pytest exception-types including Skip which all extend BaseExceptions. This should now catch everything, record it as a warning, but not ignore system interrupts.

**Pull request merged**